### ISSUE TITLE
sending to R-gui - tried but failed

### ISSUE THREAD

**Issue by mostly-harmless** [wch/SendText#11]
I used the section below in vim at somepoint and that work find to send code to the R-gui. So I tried adding something similar to sendtext (see below). However, it had no effect. Is there a way to have R-code (from an R or Rmd file) sent to the R-gui but code in an sh file sent to iterm?

let cmd = substitute(cmd, "\\", '\\\', 'g')
let cmd = substitute(cmd, '"', '\\"', "g")
let cmd = substitute(cmd, "'", "\\'", "g")
call system("osascript -e 'tell application \"R\" to cmd \"" .cmd. "\"'")

        if prog == "R":
            # Remove trailing newline
            # selection = selection.rstrip('\n')
            selection = SendSelectionCommand.escapeString(selection)

**Comment by mostly-harmless** [comment]
Alternatively, is there a way to only activate sendtext only for a particular file type?

**Comment by wch** [comment]
I think it would be possible to send to different programs based on the file type, or activate only for one type of file. If you want to take a stab at it, the API reference should be helpful:
  https://www.sublimetext.com/docs/3/api_reference.html

**Comment by mostly-harmless** [comment]
I would be happy to try but even with the API reference I wouldn't know where to start trying to update a sublime plugin. I wrote some stuff for vim at some point but there at least I could figure out how to do intermediate steps (e.g., simple step-by-step debugging). No clue how to do even that on sublime. Basically the same problem I have with java-script :)

The terminal code in sendtext works fine but the iterm functionality is giving me trouble. Line-by-line the linked file works fine. But as a whole it goes bonkers when transferred to R. I tried to isolate the problem but it seems very finicky and I can't make heads or tails of it. Mind taking a quick look to see if it produces the same problem on your machine?

https://www.dropbox.com/s/cui03fj6dhsn5t5/iterm_strange.R?dl=0
